### PR TITLE
added -f flag for formatting to rados command line tool

### PR DESCRIPTION
**Pull request by sdieffen**
Added -f flag for formatting to rados command line tool. 

Fixes: http://tracker.ceph.com/issues/15904

Signed-off-by: Steven Dieffenbach <sdieffen@redhat.com>

### CODE REVIEW

**Comment by athanatos** [comment]
The Fixes line seem to be in the wrong place.  You need a blank line between the first line (which becomes the short description in gitk etc) and the rest of the commit message.

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -3284,12 +3285,37 @@ int main(int argc, const char **argv)
...
 
-  global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_UTILITY, 0);
-  common_init_finish(g_ceph_context);
-
   std::map < std::string, std::string > opts;
   std::vector<const char*>::iterator i;
+  std::vector<const char*>::iterator j;
   std::string val;
+
+  /* Necessary to support usage of -f for formatting,
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: could use C++ style comment like 
```
// Necessary to...
```

File: src/tools/rados/rados.cc [hunk 2]
```diff
@@ -3284,12 +3285,37 @@ int main(int argc, const char **argv)
...
+   * since global_init will remove the -f using ceph
+   * argparse procedures. */
+  for (j = args.begin(); j != args.end(); ) {
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: might want to limit `j` lexical scope in the `for` loop, like
```
for (auto j = args.begin(); j != args.end(); ++j)
```

File: src/tools/rados/rados.cc [hunk 3]
```diff
@@ -3284,12 +3285,37 @@ int main(int argc, const char **argv)
...
+      unique_ptr<Formatter> formatter(Formatter::create(val.c_str()));
+      if (formatter) {
+	j = args.erase(j);
+	opts["format"] = val;
+	
+	j = args.erase(j);
+	break;
+      }
+    }
+    ++j;
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: could move this into the `for` statement.

File: src/tools/rados/rados.cc [hunk 4]
```diff
@@ -3284,12 +3285,37 @@ int main(int argc, const char **argv)
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
 
-  global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_UTILITY, 0);
-  common_init_finish(g_ceph_context);
-
   std::map < std::string, std::string > opts;
   std::vector<const char*>::iterator i;
+  std::vector<const char*>::iterator j;
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: could move the definition of `i` down to where it is used for the first time.

**Comment by tchaikov** [comment]
@sdieffen also you may need to squash these two commit into a single one.

**Comment by theanalyst** [comment]
@tchaikov this was also reported at http://tracker.ceph.com/issues/14501 (pr https://github.com/ceph/ceph/pull/7374/commits) but back then we were discussing not to defer global_init later.. but if we are are doing it, then I agree this patch is way easier than modifying global init and friends

**Pull request merged**